lib/src/ChainWebSocket.js

```diff
@@ -64,14 +65,15 @@ class ChainWebSocket {
                     if( this.send_life == 0) {
                         // this.ws.ping('', false, true);
                         if ( this.keepAliveCb ){
-                            this.keepAliveCb();
+                            this.keepAliveCb(this.closed);
                         }
                         this.send_life = max_send_life;
                     }
                 }, 5000);
                 resolve();
             }
             this.ws.onerror = (error) => {
+                this.closed = true;
```

Comment:
I don't think it's necessarily closed just because of an error happening, so this probably doesn't belong here.